src/Modules/Data/Commands.Data.Test/GetPowerBIDatasourceTests.cs

```diff
@@ -28,6 +28,12 @@ public class GetPowerBIDatasourceTests
         [TestCategory("SkipWhenLiveUnitTesting")] // Ignore for Live Unit Testing
         public void EndToEndGetPowerBIDatasourceIndividualScope()
         {
+            /*
+             * Requirement to run test:
+             * Need at least one dataset containing a datasource assigned to the user logging into the test.
+             * Update the test with the dataset ID before running.
```

Comment:
You should consider updating this and other tests to act similar to the end-to-end tests for the workspace cmdlets.

Those tests don't require you to update the test with any specific ID because they attempt to locate the first valid workspace using a shared helper method (that calls the get cmdlet). If no valid workspace is found, it returns inconclusive. After this check, it will run the intended cmdlets using the ID or workspace object.

This flow makes it very simple to run the end-to-end tests locally and figure out which scenarios weren't hit.

Reply:
Updated